core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -829,12 +867,28 @@ private[spark] class ExecutorAllocationManager(
       numTotalTasks - numRunning
     }
 
+    // Currently TaskSetManager.getCompletelyBlacklistedTaskIfAny only takes the first
+    // unschedulable task due to blacklisting. So keeping track of unschedulableTaskSets
+    // should be enough as we'll always have no more than a task unschedulable at any time.
```

Comment:
I think this comment is confusing, specifically "as we'll always have no more than a task unschedulable at any time".
That isn't true. you could have lots of tasks blacklisted and thus unschedulable at any time. That function only returns the first one it finds.